fili-core/src/main/java/com/yahoo/bard/webservice/data/config/provider/MakerDictionary.java

```diff
@@ -0,0 +1,295 @@
+                // LongSumMaker => longSum
+                String makerName = lowerFirst(cls.getSimpleName()).replaceFirst("Maker", "");
+                MetricMaker actualMaker = instantiateObjectFromArgs(cls, dict);
+                if (actualMaker != null) {
+                    defaultMakers.put(makerName, actualMaker);
+                } else {
+                    throw new ConfigurationError("Unable to instantiate built-in metric maker " + cls);
+                }
+            }
+        }
+
+        return defaultMakers;
+    }
+
+    /**
+     * Initialize the metric makers, custom and built-in.
+     *
+     * @param dict MetricDictionary to use for every maker
+     * @param mDict Dictionary of function name to MetricMaker
+     * @param makers user-defined config listing custom MetricMakers
+     */
+    public static void loadMetricMakers(
+            MetricDictionary dict,
+            MakerDictionary mDict,
+            Map<String, MakerConfiguration> makers
+    ) {
+        loadMetricMakers(dict, mDict, makers, false);
+    }
+
+    /**
+     * Initialize the metric makers, custom and optionally built-in.
+     *
+     * Eventually, 'skipDefault' will be an option in the YAML config.
+     *
+     * @param dict MetricDictionary to use for every maker
+     * @param mDict Dictionary of function name to MetricMaker
+     * @param makers user-defined config listing custom MetricMakers
+     * @param skipDefault true to skip loading built-in metrics
+     */
+    public static void loadMetricMakers(
+            MetricDictionary dict,
+            MakerDictionary mDict,
+            Map<String, MakerConfiguration> makers,
+            boolean skipDefault
+    ) {
+
+        if (!skipDefault) {
+            mDict.putAll(getDefaultMakers(dict));
+        }
+
+        if (makers != null) {
+            Map<String, MetricMaker> customMakers = makers.entrySet()
+                    .stream()
+                    .collect(Collectors.toMap(
+                            Map.Entry::getKey,
+                            v -> buildCustomMaker(
+                                    v.getKey(),
+                                    v.getValue().getClassName(),
+                                    v.getValue().getArguments(),
+                                    dict
+                            )
+                    ));
+
+            mDict.putAll(customMakers);
+        }
+    }
+
+    /**
+     * Find the full package + class name for a maker.
+     *
+     * The class name chosen for the custom maker is one of:
+     *
+     * 1. The full package + class name specified in the className field
+     * 2. The name specified in the className field inside the MetricMaker package
+     * 3. If no className given, the Maker name specified (longSum), turned into a Maker class name
+     *    (LongSumMaker), inside the MetricMaker package
+     *
+     * @param suppliedMakerName The maker name the user specified
```

Comment:
This is a good start.  Another option for number 3 would be to draw on a list of configured maker dictionaries from config.  Not needed in the first draft.